runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -565,6 +614,8 @@ function LanguageTree:_on_bytes(
   new_col,
   new_byte
 )
+  self._parser:reset()
```

Comment:
My understanding of reset was that the parser would loose all of its state. If edit is not safe after a cancellation (I thought it would be in a single threaded scenario. It worked the last time I tried cancable parsing), maybe `ts_parser_copy` would keep the state. Docs say it is meant to enable easy multithreading.

Reply:
Ahh, good point. Probably shouldn't do this.